### ISSUE TITLE
Migrations failing on master

### ISSUE THREAD

**Issue by melbourne2991** [sourcegraph/sourcegraph#5880]
- **Sourcegraph version:** Master branch
- **Platform information:** Ubuntu 16.04

#### Steps to reproduce:

1. Follow the [development setup here](https://github.com/sourcegraph/sourcegraph/blob/master/doc/dev/local_development.md#step-5-start-docker)

#### Expected behavior:
Should be able to set up a working dev environment.

#### Actual behavior:
I get a "Peer authentication failed for user" error when running `/dev/migrate.sh`

It looks as though recent changes [introduced here](https://github.com/sourcegraph/sourcegraph/commit/9f0feb4b72828c06d5677986e9d96491cc72e17c#diff-0b6554893ffee5efee8109536345a06dR13) may be responsible: 

The call to dblink_exec causes Postgres to attempt to authenticate by peer which fails if the UNIX username does not match the DB username.

I was able to get it working by running the migrations as the UNIX Postgres user and using the Postgres DB users credentials (after setting a password). [Details here](https://gist.github.com/AtulKsol/4470d377b448e56468baef85af7fd614). 

Not sure what the solution is at the moment (maybe we just need to update the readme), will revisit tomorrow morning.

**Comment by unknwon** [comment]
cc @efritz @sourcegraph/distribution

**Comment by efritz** [comment]
I couldn't reproduce with macOS using potgres from brew. What Os are you using?

**Comment by melbourne2991** [comment]
@efritz using Ubuntu. I just tested on my MacBook and it works fine for me there as well.

It looks like Postgres is installed and configured slightly differently - in Ubuntu, a Postgres admin user is created whereas in OSX the current user is used.

**Comment by melbourne2991** [comment]
Just compared the defaults for `pg_hba.conf`

Ubuntu:
```
# TYPE  DATABASE        USER            ADDRESS                 METHOD

# "local" is for Unix domain socket connections only
local   all             all                                     peer
# IPv4 local connections:
host    all             all             127.0.0.1/32            md5
# IPv6 local connections:
host    all             all             ::1/128                 md5
# Allow replication connections from localhost, by a user with the
# replication privilege.
local   replication     all                                     peer
host    replication     all             127.0.0.1/32            md5
host    replication     all             ::1/128                 md5
```
vs. OSX:
```
# TYPE  DATABASE        USER            ADDRESS                 METHOD

# "local" is for Unix domain socket connections only
local   all             all                                     trust
# IPv4 local connections:
host    all             all             127.0.0.1/32            trust
# IPv6 local connections:
host    all             all             ::1/128                 trust
# Allow replication connections from localhost, by a user with the
# replication privilege.
local   replication     all                                     trust
host    replication     all             127.0.0.1/32            trust
host    replication     all             ::1/128                 trust
```

It looks as though the default OSX config is much more open (probably because a postgres install on OSX is always going to be for development purposes) everything is just set to trust, which according to the docs means:

> When trust authentication is specified, PostgreSQL assumes that anyone who can connect to the server is authorized to access the database with whatever database user name they specify (even superuser names). Of course, restrictions made in the database and user columns still apply. This method should only be used when there is adequate operating-system-level protection on connections to the server.
https://www.postgresql.org/docs/12/auth-trust.html

I think this is why we do not see the same issue in OSX.

**Comment by efritz** [comment]
Sorry for not following up on this. This should have been fixed by #5935 a while ago.
